apps/ligo-virgo/src/pages/workspace/docs/components/toc/TOC.tsx

```diff
@@ -0,0 +1,176 @@
+import { BlockEditor } from '@toeverything/components/editor-core';
+import { styled } from '@toeverything/components/ui';
+import type { ReactNode } from 'react';
+import {
+    createContext,
+    useCallback,
+    useContext,
+    useEffect,
+    useState,
+} from 'react';
+import { useParams } from 'react-router';
+import {
+    BLOCK_TYPES,
+    destroyEventList,
+    getContentByAsyncBlocks,
+    getPageTOC,
+    listenerMap,
+    type TocType,
+} from '../../utils/toc';
+
+const StyledTOCItem = styled('a')<{ type?: string; isActive?: boolean }>(
+    ({ type, isActive }) => {
+        const common = {
+            height: '32px',
+            display: 'flex',
+            alignItems: 'center',
+            cursor: 'pointer',
+            color: isActive ? '#3E6FDB' : '#4C6275',
+        };
+
+        if (type === BLOCK_TYPES.HEADING1) {
+            return {
+                ...common,
+                padding: '0 12px',
+                fontWeight: '600',
+                fontSize: '16px',
+            };
+        }
+
+        if (type === BLOCK_TYPES.HEADING2) {
+            return {
+                ...common,
+                padding: '0 32px',
+                fontSize: '14px',
+            };
+        }
+
+        if (type === BLOCK_TYPES.HEADING3) {
+            return {
+                ...common,
+                padding: '0 52px',
+                fontSize: '12px',
+            };
+        }
+
+        if (type === BLOCK_TYPES.GROUP) {
+            return {
+                ...common,
+                margin: '6px 0px',
+                height: '46px',
+                padding: '6px 12px',
+                fontWeight: '600',
+                fontSize: '16px',
+                borderTop: '0.5px solid #E0E6EB',
+                borderBottom: '0.5px solid #E0E6EB',
+                color: isActive ? '#3E6FDB' : '#98ACBD',
+            };
+        }
+
+        return {};
+    }
+);
+
+const StyledItem = styled('div')(props => {
+    return {
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+    };
+});
+
+const TOCContext = createContext(null);
+
+interface Props {
+    children: ReactNode;
+    editor: BlockEditor;
```

Comment:
The editor is UNSTABLE now

```suggestion
    editor?: BlockEditor;
```

Reply:
get